src/App.jsx

```diff
@@ -0,0 +1,40 @@
+import { useState } from 'react';
+import '../styles/index.css';
+import AddCard from './components/templates/AddCard';
+import CardList from './components/templates/CardList';
+import SaveCard from './components/templates/SaveCard';
+import { FIRST_STEP, LAST_STEP } from './constants/constraints';
+
+function App() {
+  const [step, setStep] = useState(1);
+  const [data, setData] = useState();
+  const goToPreviousStep = () => {
+    if (step - 1 < FIRST_STEP) {
+      return;
+    }
+    setStep(step - 1);
+  };
+
+  const goToNextStep = (stepData) => {
+    if (step + 1 > LAST_STEP) {
+      return;
+    }
+    setData(stepData);
+    setStep(step + 1);
+  };
+
+  return (
+    <div className="root">
+      {step === 0 ? <CardList goToNextStep={goToNextStep} /> : null}
+      {step === 1 ? (
+        <AddCard
+          goToPreviousStep={goToPreviousStep}
+          goToNextStep={goToNextStep}
+        />
+      ) : null}
+      {step === 2 ? <SaveCard data={data} /> : null}
+    </div>
+  );
```

Comment:
```suggestion
  const Router = ({step}) => {
    switch (step){
      case 0: return <CardList goToNextStep={goToNextStep} />
      case 1: return <AddCard goToPreviousStep={goToPreviousStep} goToNextStep={goToNextStep}/>
      case 2: return <SaveCard data={data} />
      default: return null
    }
  }

  return (
    <div className="root">
      <Router step={step}/>
    </div>
  );
```
페이지 step을 상태로 관리해주셨군요! 불필요한 방어코드를 줄이고 가독성을 높이기 위해 요런식으로 변경해보는 것은 어떨까요?? (컴포넌트 내부에 컴포넌트가 있는 형태가 어색하다면 즉시 실행 함수 형태도 좋을 것 같네요!)
~~추가적으로 `react-router`에 대해 학습해보시면 많은 도움이 될 것 같습니다!(stepper가 이번 미션의 의도셨군요...!)~~

Reply:
☑️[fix: 화면 Step 전환 컴포넌트로 분리](https://github.com/gywjd/react-payments/commit/8bfd3a83f1b63bf895788489c539f3c0d5838aeb)
너무 좋은 의견주셔서 감사합니다!
`react-router`는 적용을 해두지 않아서 삼항연산자로 null return을 하는게 조금 마음에 걸렸는데,
실제 router와는 조금 다르지만 router로 분리하여 제안해주신 코드로 수정해보았습니다.